packages/medusa/src/types/product.ts

```diff
@@ -9,6 +17,68 @@ export enum ProductStatus {
   REJECTED = "rejected",
 }
 
+export class FilterableProductProps {
```

Comment:
The idea is that your endpoint schema may or may not be 1:1 with the fields that are filterable at the service level. E.g. now that we can filter on Price Lists we might want to include:

`price_list_id: string | string[]` in `GET /admin/products`. But in `GET /admin/price-lists/:id/products` we would like to exclude `price_list_id`. `FilterableProductProps` and the endpoint validation is not necessarily the same.

An improvement would be to do:

```
Pick<FilterableProductProps, "id" | ....>
```

However I don't think this is possible with classes.

Reply:
My thinking was, that we could have a `FilterableProductParams` class that would hold all the properties that are filterable in all cases, and then for each endpoint define a class that extends it. So in the case of `GET /admin/products/` it would be something like:
```
class AdminProductsParams extends FilterableProductParams {
    price_list_id: string;
}
```

and for `GET /admin/price-lists/:id/products` it would be something like:
```
class AdminPriceListsProductsParams extends FilterableProductParams {}
```

But if you find it to do more harm than good, let me know.

Reply:
Actually, I might have misunderstood something in your explanation above. Think it makes good sense to not merge filters from service level with API validation - my bad.